Upgraded to SDK 49 and I get java.lang.AssertionError: The embedded manifest is invalid or could not be read. Make sure you have configured expo-updates correctly in android/app/build.gradle on EAS built APK

### Minimal reproducible example

https://snack.expo.dev/

### Summary

I upgraded to SDK 49 and I get this error in the ADB logs from the Android app built with EAS:

`java.lang.AssertionError: The embedded manifest is invalid or could not be read. Make sure you have configured expo-updates correctly in android/app/build.gradle`

This is my app.json:

```
{
  "expo": {
    "name": "101 Soundboards",
    "slug": "sb-soundboards-all",
    "privacy": "public",
    "platforms": [
      "ios",
      "android",
      "web"
    ],
    "version": "110.0.0",
    "orientation": "default",
    "icon": "./assets/icon-android-512.png",
    "splash": {
      "image": "./assets/splash-dark-mode.png",
      "resizeMode": "contain",
      "backgroundColor": "#212529"
    },
    "updates": {
      "fallbackToCacheTimeout": 0
    },
    "assetBundlePatterns": [
      "**/*"
    ],
    "ios": {
      "supportsTablet": true,
      "bundleIdentifier": "sb.soundboards.all",
      "buildNumber": "110.0.0",
      "backgroundColor": "#212529",
      "icon": "./assets/icon-ios-1024.png",
      "isTabletOnly": false
    },
    "android": {
      "package": "sb.soundboards.all",
      "versionCode": 110,
      "blockedPermissions": [
        "android.permission.WRITE_EXTERNAL_STORAGE"
      ],
      "config": {},
      "intentFilters": [
        {
          "action": "VIEW",
          "data": [
            {
              "scheme": "https",
              "host": "*.101soundboards.com",
              "pathPrefix": "/boards/"
            }
          ],
          "category": [
            "BROWSABLE",
            "DEFAULT"
          ]
        }
      ]
    },
    "androidNavigationBar": {
      "backgroundColor": "#212529"
    },
    "scheme": "sb101",
    "hooks": {
    },
    "extra": {
      "eas": {
        "projectId": "dbac0ffc-934b-4b3d-97d3-972000582278"
      }
    },
    "plugins": [
    ]
  }
}

```

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: Windows 10 10.0.19045        
    Binaries:
      Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
      npm: 9.5.1 - C:\Program Files\nodejs\npm.CMD
    IDEs:
      Android Studio: AI-222.4459.24.2221.10121639
    npmPackages:
      expo: ^49.0.0 => 49.0.3
      react: 18.2.0 => 18.2.0
      react-dom: 18.2.0 => 18.2.0
      react-native: 0.72.3 => 0.72.3
      react-native-web: ~0.19.6 => 0.19.6
    Expo Workflow: managed

### Possible workaround

Got the app working by using `expo@49.0.6`, `expo-updates@0.18.11`, `sentry-expo@7.0.1`, and by adding the following to `package.json` as suggested [here](https://github.com/expo/expo/issues/23382#issuecomment-1626355563):

```
{ 
  // rest of the contents in package.json 
  
  "expo": {
    "autolinking": {
      "exclude": [
        "expo-updates"
      ]
    }
  }
}
```

The app has been running in production for about a week and Sentry seems to work as expected. 

### Recipe for reproducing crash on Android

1. Create a new managed Expo v49 app
2. Add Sentry by following the [offical instructions](https://docs.expo.dev/guides/using-sentry/) (with `expo-update` as a side dependency)
3. Do not configure `expo-updates`
4. Build a release version with EAS Build
5. Start release version of app from step 4 on Android and observe it crashes on bootup

For the record here is the stacktrace from Logcat:

```
2023-07-18 18:55:42.372 12968-12968 EmbeddedManifest        com.example.expocrash202307          E  Could not read embedded manifest

java.lang.NullPointerException
at expo.modules.updates.manifest.BareUpdateManifest$Companion.fromBareManifest(BareUpdateManifest.kt:97)
at expo.modules.updates.manifest.ManifestFactory.getEmbeddedManifest(ManifestFactory.kt:36)
at expo.modules.updates.manifest.EmbeddedManifest.get(EmbeddedManifest.kt:30)
at expo.modules.updates.launcher.NoDatabaseLauncher.<init>(NoDatabaseLauncher.kt:70)
at expo.modules.updates.launcher.NoDatabaseLauncher.<init>(NoDatabaseLauncher.kt:23)
at expo.modules.updates.UpdatesController.start(UpdatesController.kt:243)
at expo.modules.updates.UpdatesController$Companion.initialize(UpdatesController.kt:603)
at expo.modules.updates.UpdatesPackage$createReactNativeHostHandlers$handler$1.onWillCreateReactInstanceManager(UpdatesPackage.kt:56)
at expo.modules.ReactNativeHostWrapperBase.createReactInstanceManager(ReactNativeHostWrapperBase.kt:27)
at com.facebook.react.ReactNativeHost.getReactInstanceManager(ReactNativeHost.java:42)
at com.example.expocrash202307.MainApplication.onCreate(MainApplication.java:71)
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1277)
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6759)
at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2133)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7872)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)

2023-07-18 18:55:42.373 12968-12968 AndroidRuntime          com.example.expocrash202307          D  Shutting down VM
2023-07-18 18:55:42.373 12968-12968 AndroidRuntime          com.example.expocrash202307          E  FATAL EXCEPTION: main

Process: com.example.expocrash202307, PID: 12968
java.lang.AssertionError: The embedded manifest is invalid or could not be read. Make sure you have configured expo-updates correctly in android/app/build.gradle. null
at expo.modules.updates.manifest.EmbeddedManifest.get(EmbeddedManifest.kt:34)
at expo.modules.updates.launcher.NoDatabaseLauncher.<init>(NoDatabaseLauncher.kt:70)
at expo.modules.updates.launcher.NoDatabaseLauncher.<init>(NoDatabaseLauncher.kt:23)
at expo.modules.updates.UpdatesController.start(UpdatesController.kt:243)
at expo.modules.updates.UpdatesController$Companion.initialize(UpdatesController.kt:603)
at expo.modules.updates.UpdatesPackage$createReactNativeHostHandlers$handler$1.onWillCreateReactInstanceManager(UpdatesPackage.kt:56)
at expo.modules.ReactNativeHostWrapperBase.createReactInstanceManager(ReactNativeHostWrapperBase.kt:27)
at com.facebook.react.ReactNativeHost.getReactInstanceManager(ReactNativeHost.java:42)
at com.example.expocrash202307.MainApplication.onCreate(MainApplication.java:71)
at android.app.Instrumentation.callApplicationOnCreate(Instrumentation.java:1277)
at android.app.ActivityThread.handleBindApplication(ActivityThread.java:6759)
at android.app.ActivityThread.-$$Nest$mhandleBindApplication(Unknown Source:0)
at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2133)
at android.os.Handler.dispatchMessage(Handler.java:106)
at android.os.Looper.loopOnce(Looper.java:201)
at android.os.Looper.loop(Looper.java:288)
at android.app.ActivityThread.main(ActivityThread.java:7872)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:936)
```

--- a/docs/pages/guides/using-sentry.mdx
+++ b/docs/pages/guides/using-sentry.mdx
@@ -74,7 +74,7 @@ In your project directory, run:
 
 <Terminal
   cmd={[
-    '$ npx expo install expo-application expo-constants expo-device expo-updates @sentry/react-native',
+    '$ npx expo install expo-application expo-constants expo-device @sentry/react-native',
   ]}
 />
 

